### ISSUE TITLE
Are you accepting refactorings?

### ISSUE THREAD

**Issue by kornelski** [FirefoxGraphics/qcms#41]
The code is unidiomatic for Rust, and requires more unsafe than necessary. I'd be interested in cleaning this up, but this would result in big-ish patches, and I'd like to first ask if you have ability to accept and review bigger changes.

**Comment by jrmuizel** [comment]
Yeah, I'd probably be willing to accept larger changes to make it more idiomatic.

**Comment by jrmuizel** [comment]
If you have a rough sketch of particular changes I can pre-review that as well.

**Comment by kornelski** [comment]
I need a safe CMS library, and I was wondering whether to use qcms, or convert LCMS2 myself.

Transforms seem to have more `unsafe` code than necessary. I'd like to try to reduce amount of pointer arithmetic it performs (`src`/`dest` may be tricky due to aliasing, but other data like lookup tables could be just Rust arrays).

Plus there's a ton of leftovers from C conversion like `for i in len { vec[i] }` that could have been more idiomatic.

However, I am worried that my question didn't get an answer for 3 months, so submitting a few pull requests at that pace could take too long.

**Comment by kornelski** [comment]
A sample: #42
